dll/shellext/netplwiz/SHDisconnectNetDrives.c

```diff
@@ -0,0 +1,229 @@
+    if (dRet != NO_ERROR)
+    {
+        ERR("WNetOpenEnum failed with error: 0x%08x\n", dRet);
+        return;
+    }
+
+    lpRes = HeapAlloc(GetProcessHeap(), 0, dAllocSize);
+
+    if (lpRes == NULL)
+    {
+        ERR("Could not allocate memory.\n", dRet);
+        WNetCloseEnum(hEnum);
+        return;
+    }
+
+    do
+    {
+        DWORD dSize;
+        dCount = -1;
+        memset(lpRes, 0, dAllocSize);
+
+        dRet = WNetEnumResource(hEnum, &dCount, lpRes, &dSize);
+
+        if (dRet == WN_MORE_DATA)
+        {
+            PVOID pBuff = HeapReAlloc(GetProcessHeap(), 0, lpRes, dSize);
+
+            if (pBuff == NULL)
+            {
+                ERR("Could not reallocate memory.\n", dRet);
+                break;
+            }
+
+            lpRes = pBuff;
+            dAllocSize = dSize;
+
+            continue;
+        }
+        else if (dRet == WN_SUCCESS)
+        {
+            lpCur = lpRes;
+
+            while (dCount--)
+            {
+                memset(&item, 0, sizeof(item));
+                item.mask = LVIF_TEXT | LVIF_IMAGE;
+                item.iImage = 0;
+                item.pszText = lpCur->lpLocalName;
+                item.lParam = 0;
+                item.iItem = ListView_InsertItem(hListView, &item);
+                ListView_SetItemText(hListView, item.iItem, 1, lpCur->lpRemoteName);
+                lpCur++;
+            }
+        }
+        else
+        {
+            ERR("WNetEnumResource failed with error: 0x%08x\n", dRet);
+            break;
+        }
+    }
+    while (dRet != WN_NO_MORE_ENTRIES);
+
+    HeapFree(GetProcessHeap(), 0, lpRes);
+    WNetCloseEnum(hEnum);
+}
+
+VOID UpdateButtonStatus(WPARAM wParam, LPARAM lParam, HWND hDlg)
+{
+    LPNMHDR pnmh;
+    HWND hListView = GetDlgItem(hDlg, IDC_CONNECTEDDRIVELIST), hOkbutton = GetDlgItem(hDlg, ID_OK);
+
+    pnmh = (LPNMHDR)lParam;
+
+    if (pnmh->hwndFrom == hListView)
+    {
+        switch (pnmh->code)
+        {
+            case LVN_ITEMCHANGED:
+                if (ListView_GetSelectedCount(hListView))
+                {
+                    EnableWindow(hOkbutton, TRUE);
+                }
+                else
+                {
+                    EnableWindow(hOkbutton, FALSE);
+                }
+                break;
+        }
+    }
+}
+
+DWORD DisconnectDriveExit(HWND hDlg)
+{
+    TCHAR driveLetter[10];
+    HRESULT hr;
+    HWND hListView = GetDlgItem(hDlg, IDC_CONNECTEDDRIVELIST);
+    INT nItem = ListView_GetNextItem(hListView, -1, LVNI_SELECTED);
+
+    if (nItem == -1)
+        return 0;
+
+    ListView_GetItemText(hListView, nItem, 0, driveLetter, ARRAYSIZE(driveLetter));
+    hr = WNetCancelConnection2(driveLetter, CONNECT_UPDATE_PROFILE, FALSE);
+
+    EndDialog(hDlg, ID_OK);
+    return hr;
+}
+
+static INT_PTR CALLBACK DisconnectDlgProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    HICON hIcon = NULL, hIconSm = NULL;
+    HWND hOkbutton = GetDlgItem(hDlg, ID_OK);
```

Comment:
Please move initialization of this to the `WM_INITDIALOG` case, that is the only place where it is used.

Reply:
Same here plus I get a warning about mixing case and declarations.